### DISCUSSION TITLE
regarding CAN flashing a software to the cluster board with canoe by automation with python

### DISCUSSION THREAD

**Discussion by vnatar** [chaitu-ycr/py_canoe#46] [Ideas]
regarding CAN flashing a software to the cluster board with canoe by automation with python.is it possible to automate instead of manual key press

**Comment by chaitu-ycr** [comment]
hi @vnatar 

flashing is simply sending diagnostic requests to ECU.
if you have the entire flashing sequence then it is possible to automate flashing using this method https://chaitu-ycr.github.io/py_canoe/src_manual/#src.py_canoe.canoe.CANoe.send_diag_request

controlling key from py-canoe not possible. instead of key you can change it to system variable. it is possible to control system variable values from py-canoe.
